apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -337,8 +337,7 @@ type RollingUpdateConfig struct {
 // ClusterResourcePlacementStatus defines the observed state of the ClusterResourcePlacement object.
 type ClusterResourcePlacementStatus struct {
 	// SelectedResources contains a list of resources selected by ResourceSelectors.
-	// +optional
-	SelectedResources []ResourceIdentifier `json:"selectedResources,omitempty"`
+	SelectedResources []ResourceIdentifier `json:"selectedResources"`
```

Comment:
Is it safe to make this change if we already have v1beta1 fleet?

Reply:
we have to rollout the agent release first, and CRP controller will have 5 minutes reconcile to fulfil the selectedResources fields for the existing CRPs.

Then rollout the manifest changes to mark the field as required.

Reply:
the CRP controller is filling the array now already but the empty array got ignore. I am not sure if those are persisted in the ETCD and just not displayed or got omitted before persisting. If those are not persisted then the existing CRP will all be invalid after the CRD change.

Reply:
You're right. It will be a breaking change. Will create the CRD PR today.